### PR TITLE
Upgrade dependencies and move away from unmaintained crates

### PR DESCRIPTION
**Pull request by marcograss**
this PR upgrades the dependencies, the only one involving some changes in the code is Clap to the next major 
It also replaces memmap with memmap2 since the original crate is unmaintained

I only tested on macOS so it's better to give it a try on Windows since the windows crate has been updated as well

### CODE REVIEW

**Comment by etke** [comment]
Thanks for the updates, mind rebasing your branch on current master?

**Comment by marcograss** [comment]
> Thanks for the updates, mind rebasing your branch on current master?

done, is the CI running?

File: src/main.rs [hunk 1]
```diff
@@ -322,27 +319,25 @@ fn main() {
...
+    let procname = args.get_one::<String>("process");
+    let procall = args.get_flag("process-all");
 
-    let format = if args.is_present("json") {
-        if args.is_present("pretty") {
-            output::Format::JsonPretty
-        } else {
-            output::Format::Json
-        }
+    let format = if args.get_flag("json") {
```
Comment (cgzones) [inline src/main.rs:334]: `-j -p` would not work now

File: src/main.rs [hunk 2]
```diff
@@ -248,56 +250,47 @@ fn main() {
...
                 .help("Output in json format"),
         )
         .arg(
             Arg::new("maps")
                 .short('m')
                 .long("maps")
-                .help("Include process memory maps (Linux only)")
-                .requires("pid")
+                .action(ArgAction::SetTrue)
+                .help("Include process memory maps (linux only)")
```
Comment (cgzones) [inline src/main.rs:279]: why drop `.requires("pid")`?

**Comment by marcograss** [comment]
nice catches, I fixed them in the new commit. I got confused while rebasing

**Comment by etke** [comment]
Appears the latest windows crate added `Option` around some of the used API args https://github.com/etke/checksec.rs/actions/runs/3220737878/jobs/5285290254

If you want to update to fix those you can or you can drop the windows crate update from this PR, and I'll update the crate and associated code in a subsequent commit.

**Comment by marcograss** [comment]
I fixed the build on windows

File: Cargo.toml [hunk 3]
```diff
@@ -1,45 +1,53 @@
...
 ]
+keywords = ["checksec", "binary", "security"]
+license = "Apache-2.0"
+name = "checksec"
 readme = "README.md"
+repository = "https://github.com/etke/checksec.rs"
+version = "0.0.9"
 
 [profile.release]
-codegen-units = 1   # Reduce number of codegen units to increase optimizations
```
Comment (etke) [inline Cargo.toml:22]: Would prefer to keep comments in an aligned column for readability.

File: src/main.rs [hunk 4]
```diff
@@ -248,33 +250,33 @@ fn main() {
         .author(crate_authors!())
         .version(crate_version!())
         .arg_required_else_help(true)
-        .arg(
```
Comment (etke) [inline src/main.rs]: Args were reordered in previous commit from @cgzones, would prefer to keep alphabetical order.

File: src/main.rs [hunk 5]
```diff
@@ -306,13 +300,17 @@ fn main() {
                 .short('p')
                 .long("process")
                 .value_name("NAME")
-                .help("Name of running process to check")
-                .takes_value(true),
+                .help("Name of running process to check"),
         )
+        .arg(Arg::new("pid").long("pid").value_name("PID").help(
```
Comment (etke) [inline src/main.rs]: Please align to existing arg formatting with `.long`, `.value_name` and `.help` on separate lines.

**Review by etke — CHANGES_REQUESTED** [review]
Thank you for addressing feedback received so far, a few small formatting changes then PR is good to merge.

**Comment by marcograss** [comment]
I addressed the code formatting errors, however the one involving the Arg subcalls (help, long_name etc) might be a little fragile if cargo fmt behavior changes in the future, anyway for now it's ok

**Review by etke — APPROVED** [review]

**Comment by etke** [comment]
Merged. Thanks for your contributions!

**Pull request merged**